lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -137,6 +134,24 @@ class TraceElements extends Gatherer {
     return topFive;
   }
 
+  /**
+   * Find the node ids of elements which are animated using the Animation trace events.
+   * @param {Array<LH.TraceEvent>} mainThreadEvents
+   * @return {Array<TraceElementData>}
+   */
+  static getAnimatedElements(mainThreadEvents) {
+    const animatedElementIds = new Set(mainThreadEvents
+      .filter(e => e.name === 'Animation' && e.ph === 'b')
+      .map(e => this.getNodeIDFromTraceEvent(e)));
+
+    /** @type Array<TraceElementData> */
+    const animatedElementData = [];
+    for (const nodeId of animatedElementIds) {
+      nodeId && animatedElementData.push({nodeId});
```

Comment:
Good question.  I don't remember what compelled me to use this syntax instead of `if(nodeId) { ... }`.  I was probably in the habit after using short circuits for trace event values (e.g. `event.args && event.args.data && event.args.data.compositeFailed`).